### DISCUSSION TITLE
Autodocs Error: Check the render method of `ArgsTable`

### DISCUSSION THREAD

**Discussion by caomicc** [storybookjs/storybook#24738] [Help]
Hi!

I'm starting to implement Storybook into our UI library in our monorepo and I'm getting an error when trying to use AutoDocs - viewing the doc page(s) results in an error:
> Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
>
> Check the render method of `ArgsTable`.

Removing AutoDocs resolves the issue + I am able to publish the story without errors otherwise

I saw something about needing to change the component and how it is imported, in the following example I have tried both type of imports and see no updates:
`import { AccordionItem } from '.'`
`import AccordionItem from './AccordionItem'`

Example story:
```
import type { Meta, StoryObj } from '@storybook/react';

import { Accordion } from '@chakra-ui/react';
import AccordionItem from './AccordionItem';

const meta = {
  title: 'Magma/Atoms/AccordionItem',
  component: AccordionItem,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof AccordionItem>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  decorators: [
    (Story) => (
      <Accordion width="100vw" maxW={'5xl'} allowToggle>
        <Story />
      </Accordion>
    ),
  ],
  args: {
    heading: 'What is Workhuman Spotlight?',
  },
};

```

main.tsx
```
import type { StorybookConfig } from '@storybook/nextjs';

import { dirname, join } from 'path';

/**
 * This function is used to resolve the absolute path of a package.
 * It is needed in projects that use Yarn PnP or are set up within a monorepo.
 */
function getAbsolutePath(value: string): any {
  return dirname(require.resolve(join(value, 'package.json')));
}
const config: StorybookConfig = {
  stories: ['../**/*.mdx', '../**/*.stories.@(js|jsx|mjs|ts|tsx)'],
  typescript: {
    // Overrides the default Typescript configuration to allow multi-package components to be documented via Autodocs.
    reactDocgen: 'react-docgen',
    skipBabel: true,
    check: false,
  },
  addons: [
    getAbsolutePath('@storybook/addon-links'),
    getAbsolutePath('@storybook/addon-essentials'),
    getAbsolutePath('@storybook/addon-interactions'),
    getAbsolutePath('@chakra-ui/storybook-addon'),
  ],
  framework: {
    name: getAbsolutePath('@storybook/nextjs'),
    options: {},
  },
  previewHead: (head) => `
    ${head}
    <link rel="preconnect" href="https://fonts.googleapis.com">
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
    <link href="https://fonts.googleapis.com/css2?family=Roboto&display=swap" rel="stylesheet">
  `,
  docs: {
    autodocs: true,
  },
  staticDirs: [
    {
      from: '../fonts',
      to: 'fonts',
    },
  ],
};
export default config;
```

preview.tsx
```
import { ChakraProvider } from '@chakra-ui/react';
import type { Preview } from '@storybook/react';
import React from 'react';
import { theme } from '../theme';

const preview: Preview = {
  parameters: {
    actions: { argTypesRegex: '^on[A-Z].*' },
    controls: {
      matchers: {
        color: /(background|color)$/i,
        date: /Date$/i,
      },
    },
    chakra: {
      theme,
    },
    docs: {
      toc: true, // 👈 Enables the table of contents
    },
  },
  decorators: [
    (Story) => (
      <ChakraProvider theme={theme}>
        {/* 👇 Decorators in Storybook also accept a function. Replace <Story/> with Story() to enable it  */}
        <Story />
      </ChakraProvider>
    ),
  ],
};

export default preview;
```

Relevant package.json:
```
    "@chakra-ui/storybook-addon": "^5.0.1",
    "@storybook/addon-a11y": "^7.5.2",
    "@storybook/addon-actions": "^7.5.2",
    "@storybook/addon-essentials": "^7.5.2",
    "@storybook/addon-interactions": "^7.5.2",
    "@storybook/addon-links": "^7.5.2",
    "@storybook/addon-mdx-gfm": "^7.5.2",
    "@storybook/addon-onboarding": "^1.0.8",
    "@storybook/blocks": "^7.5.2",
    "@storybook/nextjs": "^7.5.2",
    "@storybook/react": "^7.5.2",
    "@storybook/react-docgen-typescript-plugin": "^1.0.2--canary.12.cf35881ff63bf958b8998e8dd1dfd5626ee868c2.0",
    "@storybook/testing-library": "^0.2.2",
    "storybook": "^7.5.2",
```

**Comment by gabrielbull** [comment]
Same issue here, this should be reported as an issue.

**Comment by ndelangen** [comment]
I think you're likely to have some conflicting versions of storybook packages.
- try running `sb info`.
- try regenerating your lockfile.
- try upgrading to the latest version of storybook.

**Comment by clairecassan** [comment]
Same issue today after porting from storybook 6 to 7.6.17 using the migration scripts.
If I disable all my add-ons, the error disappears, but all the generated doc is empty.
I will keep my add-ons and disable autodocs for now

**Comment by croschel** [comment]
I don't know if it is the same situation you have but in my case I changed my version to `7.4.0` and as I am using a monorepo structure I needed to use the same version of storybook in all of the projects.

**Reply by ajgolez** [reply]
Were you able to fix the issue?

**Comment by valentinpalkovic** [comment]
Can someone provide a minimal reproduction so that we can take a closer look?

**Reply by clairecassan** [reply]
![image](https://github.com/storybookjs/storybook/assets/65108365/07023a38-aafa-4851-b011-7d46c841cf1c)
[sse_cloudapp_uicomponents.zip](https://github.com/storybookjs/storybook/files/14572793/sse_cloudapp_uicomponents.zip)

**Reply by ajgolez** [reply]
Is there an update to this issue? This also happened to me when upgrading from version 6.5 to 8.

**Comment by LawrieCape** [comment]
I'm having the same issue - Upgraded from Storybook 7.4.1 to 8.0.4, went through the auto-migrations and the `sb doctor` command looks fine. I'm seeing the above error on any mdx page using `Controls` or `ArgTypes` from `@storybook/blocks` - though the controls on the default story work fine.

`sb info` gives me - 
```
  npmPackages:
    @storybook/addon-a11y: ^8.0.4 => 8.0.4 
    @storybook/addon-actions: ^8.0.4 => 8.0.4 
    @storybook/addon-essentials: ^8.0.4 => 8.0.4 
    @storybook/addon-links: ^8.0.4 => 8.0.4 
    @storybook/addon-storysource: ^8.0.4 => 8.0.4 
    @storybook/preset-scss: ^1.0.3 => 1.0.3 
```
Any pointers much appreciated.

**Reply by kyle-ahlstrom** [reply]
I'm having the same issue after upgrading our monorepo using nx. nx went from v17 to v18 and the storybook libraries went from 7.5.2 to 7.6.17. Here's sb info:

![image](https://github.com/storybookjs/storybook/assets/10853861/6b1a3392-5add-471a-918c-c1f8a14241fa)

**Reply by kyle-ahlstrom** [reply]
I was able to resolve this by manually installing @storybook/components@7.6.17. Seems like there were some old/conflicting peer dependencies that were causing the error.

**Reply by LawrieCape** [reply]
Thanks Kyle, that's fixed my issue too (along with adding the latest @storybook/theming dep as well).
